### PR TITLE
Comments extraction from PHP code

### PR DESCRIPTION
**Pull request by mlocati**
Follow-up of #100: what about a code like this?

### CODE REVIEW

File: src/Utils/PhpFunctionsScanner.php [hunk 1]
```diff
@@ -6,16 +6,55 @@
...
+    }
+
+    /**
      * Constructor.
      *
      * @param string $code The php code to scan
      */
     public function __construct($code)
     {
-        $this->tokens = token_get_all($code);
```
Comment (mlocati) [inline src/Utils/PhpFunctionsScanner.php]: Let's filter out white spaces, so that parsing token is faster

**Comment by oscarotero** [comment]
It looks fine. I think is better to extract always the comments. If the user don't want them, they can be removed later. The rest of the extractors are not configurable. What do you think?

**Comment by mlocati** [comment]
One may want only comments that start with a specific text (like xgettext), otherwise we risk to extract comments that are not related to the strings to be translated.
Think for instance at this example:

```php
// Initialize some variables
$hi = t('Hello');
$x = 1;
$y = 2;
```

So, I think we should keep the same approach as xgettext: only extract comments if requested, and optionally allow to specify a "tag" to filter comments to be extracted.

**Comment by oscarotero** [comment]
ah, so comments can be before the function, not neccesary inside?
Ok, in this case, maybe a static property (like `public static $functions`) ?

**Comment by mlocati** [comment]
Something like https://github.com/mlocati/Gettext/commit/26c3b5302640be451a589e9cff8bc35e8d70b217 ?

**Comment by oscarotero** [comment]
Its fine to me

**Comment by mlocati** [comment]
I forgot to answer this:
> ah, so comments can be before the function, not neccesary inside?

Yes, xgettext can extract comments that are imediately before the function, provided that *"comment blocks supposed to be extracted must be adjacent to keyword lines"* (from [gettext manual](https://www.gnu.org/savannah-checkouts/gnu/gettext/manual/html_node/xgettext-Invocation.html#Operation-mode))

That means that with this code

```php
<?php

echo __(/* i18n: Test comment 1 */'test 1');

/* i18n: Test comment 2 */
echo __('test 2');

/* i18n: Test comment 3 */

echo __('test 3');

/* i18n: Test comment 4 */
echo __(
    'test 4'
);
```

`xgettext` extracts comments 1, 2 and 3, but not the 4th:
```po
#. i18n: Test comment 1
#: test.php:3
msgid "test 1"
msgstr ""

#. i18n: Test comment 2
#: test.php:6
msgid "test 2"
msgstr ""

#. i18n: Test comment 3
#: test.php:10
msgid "test 3"
msgstr ""

#: test.php:14
msgid "test 4"
msgstr ""
```

The current implementation of this pull request is a bit different: it will extracts also `i18n: Test comment 4`, but I don't think this will be a big problem.

**Comment by mlocati** [comment]
Ok, I'll add a few tests and everything is ready to be merged :wink:

**Comment by mlocati** [comment]
Done.

**Comment by mlocati** [comment]
After this has been merged I'll stop boring you and I'll be looking forward to a new release :wink:

**Comment by oscarotero** [comment]
Thanks for your hard work. I'm going to release the new 3.5.9 version soon.

**Comment by mlocati** [comment]
> Thanks for your hard work.

Thanks to you!

> I'm going to release the new 3.5.9 version soon.

Great!

**Comment by oscarotero** [comment]
The wait is over: https://github.com/oscarotero/Gettext/releases/tag/v3.5.9

**Comment by mlocati** [comment]
Super fast! Thank you!!

**Pull request merged**